src/sci/impl/namespaces.cljc

```diff
@@ -1011,6 +1011,7 @@
    #?@(:clj ['proxy* (core-var 'proxy* proxy/proxy* true)
              'proxy (macrofy 'proxy proxy/proxy clojure-core-ns true)])
    'satisfies? (copy-core-var sci.impl.protocols/satisfies?)
+   'type->str (copy-core-var sci.impl.protocols/type->str)
```

Comment:
Why do we need this one?

Reply:
Ah, I didn't realize CLJS has `type->str`. Should we also "fix" it to properly return `"string"` for `js/String`, something like:

```
(.-name (.-constructor (.-prototype js/String)))
```

?

Reply:
the implementation of this almost exactly matches how `type->str` works in CLJS

```
ClojureScript 1.11.57
cljs.user=> (type->str js/String)
"function String() { [native code] }"
cljs.user=>
```

Reply:
I also detail why this is needed specifically for fixing this bug in the PR description, copied below

> In addition, it also exposes a new var clojure.core/type->str. The rationale for it is we need to use the same machinery in two different contexts: a macro context in extend-type and extend-protocol, and at runtime in satisfies?. I could emit the code directly in the macro context instead, but I thought it would be better for maintenance purposes to have everything refer to a single function.